From 3f9c8abbb6e4fa504c71a6238e58c5e3ac1f4fc4 Mon Sep 17 00:00:00 2001
From: Bharat Bhushan <Bharat.Bhushan@freescale.com>
Date: Thu, 26 Nov 2015 20:52:31 +0530
Subject: [PATCH 343/452] vfio fsl-mc: Make vfio irq handler non-threaded

For performance we need to make vfio handler non-threaded.
With this we see a lot more improved performance with most
of networking benchmarks.

Signed-off-by: Bharat Bhushan <Bharat.Bhushan@freescale.com>
[Xulin: Original patch taken from FSL LS2085 SDK EAR6.0,
LS2085A-SDK-SOURCE-20160304-yocto.iso]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/vfio/fsl-mc/vfio_fsl_mc_intr.c |   12 ++++--------
 1 files changed, 4 insertions(+), 8 deletions(-)

diff --git a/drivers/vfio/fsl-mc/vfio_fsl_mc_intr.c b/drivers/vfio/fsl-mc/vfio_fsl_mc_intr.c
index bc60e86..e33d205 100644
--- a/drivers/vfio/fsl-mc/vfio_fsl_mc_intr.c
+++ b/drivers/vfio/fsl-mc/vfio_fsl_mc_intr.c
@@ -295,7 +295,7 @@ static int vfio_fsl_mc_disable_irq(struct fsl_mc_device *mc_dev, int irq_num)
 	return error;
 }
 
-static irqreturn_t vfio_threaded_irq_handler(int irq_num, void *arg)
+static irqreturn_t vfio_fsl_mc_irq_handler(int irq_num, void *arg)
 {
 	struct vfio_fsl_mc_device *vdev;
 	int i;
@@ -331,15 +331,11 @@ int vfio_fsl_mc_configure_irq(struct vfio_fsl_mc_device *vdev,
 	if (WARN_ON(mc_irq->irq_configured))
 		return -EINVAL;
 
-	error = request_threaded_irq(irq->irq_number, NULL,
-				     vfio_threaded_irq_handler,
-				     IRQF_NO_SUSPEND | IRQF_ONESHOT,
-				     "VFIO Handler",
-				     vdev);
+	error = request_irq(irq->irq_number, vfio_fsl_mc_irq_handler,
+			    0, "VFIO Handler", vdev);
 	if (error < 0) {
 		dev_err(&mc_dev->dev,
-			"devm_request_threaded_irq() failed: %d\n",
-			error);
+			"IRQ registration fails with error: %d\n", error);
 		return error;
 	}
 
-- 
1.7.5.4

